sycl/test-e2e/ESIMD/lsc/lsc_local_accessor_gather_scatter.cpp

```diff
@@ -0,0 +1,121 @@
+//==-- lsc_local_accessor_gather_scatter.cpp - DPC++ ESIMD on-device test -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu-intel-pvc
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+//
+// TODO: Enable the test when GPU driver is ready/fixed.
+// XFAIL: opencl || windows || gpu-intel-pvc
+// TODO: add support for local_accessors to esimd_emulator.
+// UNSUPPORTED: esimd_emulator
+// The test checks functionality of the gather/scatter local
+// accessor-based ESIMD intrinsics.
+
+#include "../esimd_test_utils.hpp"
+
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::intel::experimental::esimd;
+
+template <typename T, unsigned VL, unsigned STRIDE> bool test(queue q) {
+  constexpr size_t size = VL;
+  constexpr int MASKED_LANE = VL - 1;
+
+  std::cout << "Testing T=" << typeid(T).name() << " VL=" << VL
+            << " STRIDE=" << STRIDE << "...\n";
+
+  // The test is going to use size elements of T type.
+  auto Dev = q.get_device();
+  auto DeviceSLMSize = Dev.get_info<sycl::info::device::local_mem_size>();
+  if (DeviceSLMSize < size) {
+    // Report an error - the test needs a fix.
+    std::cerr << "Error: Test needs more SLM memory than device has!"
+              << std::endl;
+    return false;
+  }
+
+  T *A = new T[size];
+
+  for (unsigned i = 0; i < size; ++i) {
+    A[i] = static_cast<T>(i);
+  }
+
+  try {
+    buffer<T, 1> buf(A, range<1>(size));
+    nd_range<1> glob_range{range<1>{1}, range<1>{1}};
+
+    q.submit([&](handler &cgh) {
+       auto acc = buf.template get_access<access::mode::read_write>(cgh);
+       auto LocalAcc = local_accessor<T, 1>(size * STRIDE, cgh);
+       cgh.parallel_for(glob_range, [=](id<1> i) SYCL_ESIMD_KERNEL {
+         using namespace sycl::ext::intel::esimd;
+         simd<T, VL> valsIn;
+         valsIn.copy_from(acc, 0);
+         simd_mask<VL> pred = 1;
+         pred[MASKED_LANE] = 0; // mask out the last lane
```

Comment:
For my education, is there a specific reason for special treatment of the last lane?